fix golint issues

**Is your feature request related to a problem?/Why is this needed**
<!-- A clear and concise description of what the problem is. Ex. I'm always frustrated when [...] -->

**Describe the solution you'd like in detail**
<!-- A clear and concise description of what you want to happen. -->

Also could enable `golangci-lint run --no-config --enable=golint --disable=typecheck --deadline=10m` for `hack/verify-golint.sh` script
```
# golangci-lint run --no-config --enable=golint --disable=typecheck --deadline=10m
test/utils/credentials/credentials.go:67:6: type name will be used as credentials.CredentialsConfig by other packages, and that stutters; consider calling this Config (golint)
type CredentialsConfig struct {
     ^
test/utils/credentials/credentials.go:73:6: type name will be used as credentials.CredentialsFromProw by other packages, and that stutters; consider calling this FromProw (golint)
type CredentialsFromProw struct {
     ^
test/utils/azure/azure_helpers.go:35:6: type name will be used as azure.AzureClient by other packages, and that stutters; consider calling this Client (golint)
type AzureClient struct {
     ^
pkg/azuredisk/azure_common_linux.go:75:2: don't use underscores in Go names; var `scsi_path` should be `scsiPath` (golint)
        scsi_path := "/sys/class/scsi_host/"
        ^
pkg/azuredisk/azure_common_linux.go:101:2: don't use underscores in Go names; var `sys_path` should be `sysPath` (golint)
        sys_path := "/sys/bus/scsi/devices"
        ^
pkg/azuredisk/azuredisk.go:253:6: func getSourceVolumeId should be getSourceVolumeID (golint)
func getSourceVolumeId(snapshot *compute.Snapshot) string {
     ^
pkg/azuredisk/azuredisk.go:290:10: `if` block ends with a `return` statement, so drop this `else` and outdent its block (golint)
                } else {
                       ^
pkg/azuredisk/azuredisk_test.go:391:2: var `sourceResourceSnapshotId` should be `sourceResourceSnapshotID` (golint)
        sourceResourceSnapshotId := "/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.Compute/snapshots/xxx"
        ^
pkg/azuredisk/azuredisk_test.go:392:2: var `sourceResourceVolumeId` should be `sourceResourceVolumeID` (golint)
        sourceResourceVolumeId := "/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.Compute/disks/xxx"
        ^
pkg/azuredisk/controllerserver.go:610:10: `if` block ends with a `return` statement, so drop this `else` and outdent its block (golint)
                } else {
                       ^
pkg/azuredisk/nodeserver.go:52:9: `if` block ends with a `return` statement, so drop this `else` and outdent its block (golint)
        } else {
               ^
```

**Describe alternatives you've considered**
<!-- A clear and concise description of any alternative solutions or features you've considered. -->

**Additional context**
<!-- Add any other context or screenshots about the feature request here. -->

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -1,0 +1,8 @@
+run:
+  deadline: 10m
+
+linters:
+  disable:
+  - typecheck
+  enable:
+  - golint

--- a/hack/verify-golint.sh
+++ b/hack/verify-golint.sh
@@ -21,6 +21,8 @@ if [[ -z "$(command -v golangci-lint)" ]]; then
   curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.24.0
 fi
 
-golangci-lint run --deadline=10m
+readonly PKG_ROOT="$(git rev-parse --show-toplevel)"
 
-echo "Congratulations! Lint check completed for all Go source files."
+golangci-lint run --config ${PKG_ROOT}/.golangci.yml
+
+echo "Success! Lint check completed for all Go source files."

--- a/pkg/azuredisk/azure_common_linux.go
+++ b/pkg/azuredisk/azure_common_linux.go
@@ -72,17 +72,17 @@ func getDiskLinkByDevName(io ioHandler, devLinkPath, devName string) (string, er
 }
 
 func scsiHostRescan(io ioHandler, m *mount.SafeFormatAndMount) {
-	scsi_path := "/sys/class/scsi_host/"
-	if dirs, err := io.ReadDir(scsi_path); err == nil {
+	scsiPath := "/sys/class/scsi_host/"
+	if dirs, err := io.ReadDir(scsiPath); err == nil {
 		for _, f := range dirs {
-			name := scsi_path + f.Name() + "/scan"
+			name := scsiPath + f.Name() + "/scan"
 			data := []byte("- - -")
 			if err = io.WriteFile(name, data, 0666); err != nil {
 				klog.Warningf("failed to rescan scsi host %s", name)
 			}
 		}
 	} else {
-		klog.Warningf("failed to read %s, err %v", scsi_path, err)
+		klog.Warningf("failed to read %s, err %v", scsiPath, err)
 	}
 }
 
@@ -98,8 +98,8 @@ func formatAndMount(source, target, fstype string, options []string, m *mount.Sa
 // finds a device mounted to "current" node
 func findDiskByLunWithConstraint(lun int, io ioHandler, azureDisks []string) (string, error) {
 	var err error
-	sys_path := "/sys/bus/scsi/devices"
-	if dirs, err := io.ReadDir(sys_path); err == nil {
+	sysPath := "/sys/bus/scsi/devices"
+	if dirs, err := io.ReadDir(sysPath); err == nil {
 		for _, f := range dirs {
 			name := f.Name()
 			// look for path like /sys/bus/scsi/devices/3:0:0:1
@@ -131,7 +131,7 @@ func findDiskByLunWithConstraint(lun int, io ioHandler, azureDisks []string) (st
 			if lun == l {
 				// find the matching LUN
 				// read vendor and model to ensure it is a VHD disk
-				vendorPath := filepath.Join(sys_path, name, "vendor")
+				vendorPath := filepath.Join(sysPath, name, "vendor")
 				vendorBytes, err := io.ReadFile(vendorPath)
 				if err != nil {
 					klog.Errorf("failed to read device vendor, err: %v", err)
@@ -143,7 +143,7 @@ func findDiskByLunWithConstraint(lun int, io ioHandler, azureDisks []string) (st
 					continue
 				}
 
-				modelPath := filepath.Join(sys_path, name, "model")
+				modelPath := filepath.Join(sysPath, name, "model")
 				modelBytes, err := io.ReadFile(modelPath)
 				if err != nil {
 					klog.Errorf("failed to read device model, err: %v", err)
@@ -156,7 +156,7 @@ func findDiskByLunWithConstraint(lun int, io ioHandler, azureDisks []string) (st
 				}
 
 				// find a disk, validate name
-				dir := filepath.Join(sys_path, name, "block")
+				dir := filepath.Join(sysPath, name, "block")
 				if dev, err := io.ReadDir(dir); err == nil {
 					found := false
 					devName := dev[0].Name()

--- a/pkg/azuredisk/azuredisk.go
+++ b/pkg/azuredisk/azuredisk.go
@@ -250,7 +250,7 @@ func checkDiskName(diskName string) bool {
 	return true
 }
 
-func getSourceVolumeId(snapshot *compute.Snapshot) string {
+func getSourceVolumeID(snapshot *compute.Snapshot) string {
 	if snapshot != nil &&
 		snapshot.SnapshotProperties != nil &&
 		snapshot.SnapshotProperties.CreationData != nil &&
@@ -287,9 +287,9 @@ func getValidCreationData(subscriptionID, resourceGroup, sourceResourceID, sourc
 	if len(splits) > 9 {
 		if sourceType == sourceSnapshot {
 			return compute.CreationData{}, fmt.Errorf("sourceResourceID(%s) is invalid, correct format: %s", sourceResourceID, diskSnapshotPathRE)
-		} else {
-			return compute.CreationData{}, fmt.Errorf("sourceResourceID(%s) is invalid, correct format: %s", sourceResourceID, managedDiskPathRE)
 		}
+
+		return compute.CreationData{}, fmt.Errorf("sourceResourceID(%s) is invalid, correct format: %s", sourceResourceID, managedDiskPathRE)
 	}
 	return compute.CreationData{
 		CreateOption:     compute.Copy,

--- a/pkg/azuredisk/azuredisk_test.go
+++ b/pkg/azuredisk/azuredisk_test.go
@@ -338,7 +338,7 @@ func TestCheckDiskName(t *testing.T) {
 	}
 }
 
-func TestGetSourceVolumeId(t *testing.T) {
+func TestGetSourceVolumeID(t *testing.T) {
 	SourceResourceID := "test"
 
 	tests := []struct {
@@ -380,7 +380,7 @@ func TestGetSourceVolumeId(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		result := getSourceVolumeId(test.snapshot)
+		result := getSourceVolumeID(test.snapshot)
 		if !reflect.DeepEqual(result, test.expected) {
 			t.Errorf("input: %v, getValidFileShareName result: %q, expected: %q", test.snapshot, result, test.expected)
 		}
@@ -388,8 +388,8 @@ func TestGetSourceVolumeId(t *testing.T) {
 }
 
 func TestGetValidCreationData(t *testing.T) {
-	sourceResourceSnapshotId := "/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.Compute/snapshots/xxx"
-	sourceResourceVolumeId := "/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.Compute/disks/xxx"
+	sourceResourceSnapshotID := "/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.Compute/snapshots/xxx"
+	sourceResourceVolumeID := "/subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.Compute/disks/xxx"
 
 	tests := []struct {
 		subscriptionID   string
@@ -416,7 +416,7 @@ func TestGetValidCreationData(t *testing.T) {
 			sourceType:       sourceSnapshot,
 			expected1: compute.CreationData{
 				CreateOption:     compute.Copy,
-				SourceResourceID: &sourceResourceSnapshotId,
+				SourceResourceID: &sourceResourceSnapshotID,
 			},
 			expected2: nil,
 		},
@@ -427,7 +427,7 @@ func TestGetValidCreationData(t *testing.T) {
 			sourceType:       sourceSnapshot,
 			expected1: compute.CreationData{
 				CreateOption:     compute.Copy,
-				SourceResourceID: &sourceResourceSnapshotId,
+				SourceResourceID: &sourceResourceSnapshotID,
 			},
 			expected2: nil,
 		},
@@ -480,7 +480,7 @@ func TestGetValidCreationData(t *testing.T) {
 			sourceType:       sourceVolume,
 			expected1: compute.CreationData{
 				CreateOption:     compute.Copy,
-				SourceResourceID: &sourceResourceVolumeId,
+				SourceResourceID: &sourceResourceVolumeID,
 			},
 			expected2: nil,
 		},
@@ -491,7 +491,7 @@ func TestGetValidCreationData(t *testing.T) {
 			sourceType:       sourceVolume,
 			expected1: compute.CreationData{
 				CreateOption:     compute.Copy,
-				SourceResourceID: &sourceResourceVolumeId,
+				SourceResourceID: &sourceResourceVolumeID,
 			},
 			expected2: nil,
 		},

--- a/pkg/azuredisk/controllerserver.go
+++ b/pkg/azuredisk/controllerserver.go
@@ -607,9 +607,9 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
 	if rerr != nil {
 		if strings.Contains(rerr.Error().Error(), "existing disk") {
 			return nil, status.Error(codes.AlreadyExists, fmt.Sprintf("request snapshot(%s) under rg(%s) already exists, but the SourceVolumeId is different, error details: %v", snapshotName, d.cloud.ResourceGroup, rerr.Error()))
-		} else {
-			return nil, status.Error(codes.Internal, fmt.Sprintf("create snapshot error: %v", rerr.Error()))
 		}
+
+		return nil, status.Error(codes.Internal, fmt.Sprintf("create snapshot error: %v", rerr.Error()))
 	}
 	klog.V(2).Infof("create snapshot(%s) under rg(%s) successfully", snapshotName, d.cloud.ResourceGroup)
 
@@ -775,7 +775,7 @@ func generateCSISnapshot(sourceVolumeID string, snapshot *compute.Snapshot) (*cs
 	}
 
 	if sourceVolumeID == "" {
-		sourceVolumeID = getSourceVolumeId(snapshot)
+		sourceVolumeID = getSourceVolumeID(snapshot)
 	}
 
 	return &csi.Snapshot{
@@ -846,7 +846,7 @@ func getEntriesAndNextToken(req *csi.ListSnapshotsRequest, snapshots []compute.S
 	match := false
 	entries := []*csi.ListSnapshotsResponse_Entry{}
 	for i, snapshot := range snapshots {
-		if req.SourceVolumeId == getSourceVolumeId(&snapshot) {
+		if req.SourceVolumeId == getSourceVolumeID(&snapshot) {
 			match = true
 		}
 		csiSnapshot, err := generateCSISnapshot(req.SourceVolumeId, &snapshot)

--- a/pkg/azuredisk/nodeserver.go
+++ b/pkg/azuredisk/nodeserver.go
@@ -49,9 +49,9 @@ const (
 func getDefaultFsType() string {
 	if runtime.GOOS == "windows" {
 		return defaultWindowsFsType
-	} else {
-		return defaultLinuxFsType
 	}
+
+	return defaultLinuxFsType
 }
 
 // NodeStageVolume mount disk device to a staging path

--- a/test/utils/azure/azure_helpers.go
+++ b/test/utils/azure/azure_helpers.go
@@ -32,7 +32,7 @@ import (
 	"github.com/Azure/go-autorest/autorest/to"
 )
 
-type AzureClient struct {
+type Client struct {
 	environment    azure.Environment
 	subscriptionID string
 	groupsClient   resources.GroupsClient
@@ -42,7 +42,7 @@ type AzureClient struct {
 	vnetClient     network.VirtualNetworksClient
 }
 
-func GetAzureClient(cloud, subscriptionID, clientID, tenantID, clientSecret string) (*AzureClient, error) {
+func GetAzureClient(cloud, subscriptionID, clientID, tenantID, clientSecret string) (*Client, error) {
 	env, err := azure.EnvironmentFromName(cloud)
 	if err != nil {
 		return nil, err
@@ -61,7 +61,7 @@ func GetAzureClient(cloud, subscriptionID, clientID, tenantID, clientSecret stri
 	return getClient(env, subscriptionID, tenantID, armSpt), nil
 }
 
-func (az *AzureClient) EnsureResourceGroup(ctx context.Context, name, location string, managedBy *string) (resourceGroup *resources.Group, err error) {
+func (az *Client) EnsureResourceGroup(ctx context.Context, name, location string, managedBy *string) (resourceGroup *resources.Group, err error) {
 	var tags map[string]*string
 	group, err := az.groupsClient.Get(ctx, name)
 	if err == nil && group.Tags != nil {
@@ -87,7 +87,7 @@ func (az *AzureClient) EnsureResourceGroup(ctx context.Context, name, location s
 	return &response, nil
 }
 
-func (az *AzureClient) DeleteResourceGroup(ctx context.Context, groupName string) error {
+func (az *Client) DeleteResourceGroup(ctx context.Context, groupName string) error {
 	_, err := az.groupsClient.Get(ctx, groupName)
 	if err == nil {
 		future, err := az.groupsClient.Delete(ctx, groupName)
@@ -104,7 +104,7 @@ func (az *AzureClient) DeleteResourceGroup(ctx context.Context, groupName string
 	return nil
 }
 
-func (az *AzureClient) EnsureVirtualMachine(ctx context.Context, groupName, location, vmName string) (vm compute.VirtualMachine, err error) {
+func (az *Client) EnsureVirtualMachine(ctx context.Context, groupName, location, vmName string) (vm compute.VirtualMachine, err error) {
 	nic, err := az.EnsureNIC(ctx, groupName, location, vmName+"-nic", vmName+"-vnet", vmName+"-subnet")
 	if err != nil {
 		return vm, err
@@ -158,7 +158,7 @@ func (az *AzureClient) EnsureVirtualMachine(ctx context.Context, groupName, loca
 	return future.Result(az.vmClient)
 }
 
-func (az *AzureClient) EnsureNIC(ctx context.Context, groupName, location, nicName, vnetName, subnetName string) (nic network.Interface, err error) {
+func (az *Client) EnsureNIC(ctx context.Context, groupName, location, nicName, vnetName, subnetName string) (nic network.Interface, err error) {
 	_, err = az.EnsureVirtualNetworkAndSubnet(ctx, groupName, location, vnetName, subnetName)
 	if err != nil {
 		return nic, err
@@ -201,7 +201,7 @@ func (az *AzureClient) EnsureNIC(ctx context.Context, groupName, location, nicNa
 	return future.Result(az.nicClient)
 }
 
-func (az *AzureClient) EnsureVirtualNetworkAndSubnet(ctx context.Context, groupName, location, vnetName, subnetName string) (vnet network.VirtualNetwork, err error) {
+func (az *Client) EnsureVirtualNetworkAndSubnet(ctx context.Context, groupName, location, vnetName, subnetName string) (vnet network.VirtualNetwork, err error) {
 	future, err := az.vnetClient.CreateOrUpdate(
 		ctx,
 		groupName,
@@ -235,7 +235,7 @@ func (az *AzureClient) EnsureVirtualNetworkAndSubnet(ctx context.Context, groupN
 	return future.Result(az.vnetClient)
 }
 
-func (az *AzureClient) GetVirtualNetworkSubnet(ctx context.Context, groupName, vnetName, subnetName string) (network.Subnet, error) {
+func (az *Client) GetVirtualNetworkSubnet(ctx context.Context, groupName, vnetName, subnetName string) (network.Subnet, error) {
 	return az.subnetsClient.Get(ctx, groupName, vnetName, subnetName, "")
 }
 
@@ -248,8 +248,8 @@ func getOAuthConfig(env azure.Environment, subscriptionID, tenantID string) (*ad
 	return oauthConfig, nil
 }
 
-func getClient(env azure.Environment, subscriptionID, tenantID string, armSpt *adal.ServicePrincipalToken) *AzureClient {
-	c := &AzureClient{
+func getClient(env azure.Environment, subscriptionID, tenantID string, armSpt *adal.ServicePrincipalToken) *Client {
+	c := &Client{
 		environment:    env,
 		subscriptionID: subscriptionID,
 		groupsClient:   resources.NewGroupsClientWithBaseURI(env.ResourceManagerEndpoint, subscriptionID),

--- a/test/utils/credentials/credentials.go
+++ b/test/utils/credentials/credentials.go
@@ -62,15 +62,15 @@ const (
 	locationChinaEnvVar        = "LOCATION_CHINA"
 )
 
-// CredentialsConfig is used in Prow to store Azure credentials
+// ProwConfig is used in Prow to store Azure credentials
 // https://github.com/kubernetes/test-infra/blob/master/kubetest/azure.go#L116-L118
-type CredentialsConfig struct {
-	Creds CredentialsFromProw
+type ProwConfig struct {
+	Creds ProwCredentials
 }
 
-// CredentialsFromProw is used in Prow to store Azure credentials
+// ProwCredentials is used in Prow to store Azure credentials
 // https://github.com/kubernetes/test-infra/blob/master/kubetest/azure.go#L107-L114
-type CredentialsFromProw struct {
+type ProwCredentials struct {
 	ClientID           string
 	ClientSecret       string
 	TenantID           string
@@ -157,14 +157,14 @@ func DeleteAzureCredentialFile() error {
 
 // getCredentialsFromAzureCredentials parses the azure credentials toml (AZURE_CREDENTIALS)
 // in Prow and return the credential information usable to Azure File CSI driver
-func getCredentialsFromAzureCredentials(azureCredentialsPath string) (*CredentialsFromProw, error) {
+func getCredentialsFromAzureCredentials(azureCredentialsPath string) (*ProwCredentials, error) {
 	content, err := ioutil.ReadFile(azureCredentialsPath)
 	klog.V(2).Infof("Reading credentials file %v", azureCredentialsPath)
 	if err != nil {
 		return nil, fmt.Errorf("error reading credentials file %v %v", azureCredentialsPath, err)
 	}
 
-	c := CredentialsConfig{}
+	c := ProwConfig{}
 	if err := toml.Unmarshal(content, &c); err != nil {
 		return nil, fmt.Errorf("error parsing credentials file %v %v", azureCredentialsPath, err)
 	}

